core/include/traccc/ambiguity_resolution/greedy_ambiguity_resolution_algorithm.hpp

```diff
@@ -156,6 +159,10 @@ class greedy_ambiguity_resolution_algorithm
         state_t& final_state) const;
 
     config_t _config;
+
+    std::unique_ptr<const Logger> m_logger;
+
+    const Logger& logger() const override { return *m_logger; }
```

Comment:
Let's talk in the afternoon in person. Since I still very much dislike the amount of boilerplate code added in this PR. :frowning: Since this is something that we'll keep as a design definitely for a while, I do want to come to something that we can all live with...

Reply:
This is the ACTS design, if we want to use ACTS_XXX macros, this is what it'll be.

Reply:
There **has to** be a way of factorizing this with a mix-in class. I have a very hard time accepting that every single client has to have their own member functions defined separately. We stopped doing that in Athena a decade ago...